src/content/insertable-streams/webgpu/js/multi-video.js

```diff
@@ -0,0 +1,245 @@
+        let videoFrame, videoFrame2;
+        if (frame) {
+            videoFrame = await createImageBitmap(frame, { resizeWidth: 480, resizeHeight: 270 });
+            device.queue.copyExternalImageToTexture(
+                { source: videoFrame, origin: { x: 0, y: 0 } },
+                { texture: videoTexture, origin: { x: 0, y: 270 } },
+                {
+                    // the width of the image being copied
+                    width: videoFrame.width,
+                    height: videoFrame.height,
+                }
+            );
+            videoFrame.close();
+            frame.close();
+        }
+        if (frame2) {
+            videoFrame2 = await createImageBitmap(frame2, { resizeWidth: 480, resizeHeight: 270 });
+            device.queue.copyExternalImageToTexture(
+                { source: videoFrame2, origin: { x: 0, y: 0 } },
+                { texture: videoTexture, origin: { x: 480, y: 0 } },
+                {
+                    width: videoFrame2.width,
+                    height: videoFrame2.height,
+                }
+            );
+            videoFrame2.close();
+            frame2.close();
+        }
+        // const renderPipeline = this.renderPipeline_;
+        const uniformBindGroup = device.createBindGroup({
+            layout: this.renderPipeline_.getBindGroupLayout(0),
+            entries: [
+                {
+                    binding: 0,
+                    resource: sampler,
+                },
+                {
+                    binding: 1,
+                    resource: videoTexture.createView(),
+                },
+            ],
+        });
+
+        const commandEncoder = device.createCommandEncoder();
+        const textureView = this.context_.getCurrentTexture().createView();
+
+        const renderPassDescriptor = {
+            colorAttachments: [
+                {
+                    view: textureView,
+                    loadValue: { r: 0.0, g: 0.0, b: 0.0, a: 1.0 },
+                    storeOp: 'store',
+                },
+            ],
+        };
+        // const verticesBuffer = this.verticesBuffer_;
+        const passEncoder = commandEncoder.beginRenderPass(renderPassDescriptor);
+        passEncoder.setPipeline(this.renderPipeline_);
+        passEncoder.setVertexBuffer(0, this.verticesBuffer_);
+        passEncoder.setBindGroup(0, uniformBindGroup);
+        passEncoder.draw(6, 1, 0, 0);
+        passEncoder.endPass();
+        device.queue.submit([commandEncoder.finish()]);
+    }
+
+    /** @override */
+    async destroy() {
+        if (this.device_) {
+            // Not yet in canary
+            // await this.device_.destroy();
+            this.verticesBuffer_.destroy();
+            this.device_ = null;
+            if (this.canvas_.parentNode) {
+                this.canvas_.parentNode.removeChild(this.canvas_);
+            }
+            console.log('[WebGPUTransform] WebGPU context is lost.',);
```

Comment:
nit: I think maybe this log message is copy/pasted? Maybe something like "WebGPUTransform destroyed"?